### ISSUE TITLE
Question on training Yolo v1

### ISSUE THREAD

**Issue by leonardoaraujosantos** [ruiminshen/yolo-tf#2]
Hi @ruiminshen, thanks for the project.

Could you show me how should I train for the yolo v1 (Tiny)?
Do I need to change the config.ini file? (I already downloaded the datasets VOC)

Thanks

**Comment by leonardoaraujosantos** [comment]
What I'm trying to do....

1. Download VOC 2007/2012, and extract at: Documents/Database/VOCdevki
2. python3 cache.py -c config.ini config/yolo/tiny-20.ini -v
3. python train.py -c config.ini config/yolo/tiny-20.ini

**Comment by leonardoaraujosantos** [comment]
Hi @ruiminshen I think is working
![train](https://cloud.githubusercontent.com/assets/898383/26269059/87de2906-3cea-11e7-80a3-75fa92607af3.png)

Btw I'm documenting my findings about Yolo here:
https://leonardoaraujosantos.gitbooks.io/artificial-inteligence/content/single-shot-detectors/yolo.html

Do you accept a pull request

**Comment by ruiminshen** [comment]
@leonardoaraujosantos Sorry, I'm a little busy these days and I didn't noticed your message.

Your operation is correct. To switch between models (YOLOv1 and YOLOv2), you can change the option "model" in "config" section in the default configuration file "config.ini", and the -c argument is used to override the options by another configuration file.

**Comment by xhzcyc** [comment]
When I trained it by following leonardoaraujosantos, I got this problem.
WARNING:tensorflow:Error encountered when serializing LAYER_NAME_UIDS.
Type is unsupported, or the types of the items don't match field type in CollectionDef.
'dict' object has no attribute 'name'

**Comment by TaihuLight** [comment]
Do you train successfully with YOLOv2?

**Comment by zhyj3038** [comment]
@TaihuLight   hi?  can you get a accurate resuklt with yolo2?

**Comment by TaihuLight** [comment]
@zhyj3038 I can get the mAP with YOLO2, but multi-scale trianing and training with own dataset are need to be finished! I have mail you.

**Comment by simo23** [comment]
Hi everyone! First of all thanks @ruiminshen for the amazing work. 

I've been studying your code for a while now and I could not get the training work when using the yolov2 tiny model. In particular I wanted to ask you about the yolov2 loss function. When you are extracting the values just computed by the forward pass here:

`` inputs = tf.reshape(net, [-1, cells, len(anchors), 5 + classes], name='inputs')
        with tf.name_scope('regress'):
            with tf.name_scope('inputs'):
                with tf.name_scope('inputs_sigmoid') as name:
                    inputs_sigmoid = tf.nn.sigmoid(inputs[:, :, :, :3], name=name)
                self.iou = tf.identity(inputs_sigmoid[:, :, :, 0], name='iou')
                self.offset_xy = tf.identity(inputs_sigmoid[:, :, :, 1:3], name='offset_xy')
                with tf.name_scope('wh') as name:
                    self.wh = tf.identity(tf.exp(inputs[:, :, :, 3:5]) * np.reshape(anchors, [1, 1, len(anchors), -1]), name=name)
                with tf.name_scope('prob') as name:
self.prob = tf.identity(tf.nn.softmax(inputs[:, :, :, 5:]), `name=name)``

Are you sure about the values that you are picking with inputs[:,:,:,values]? 

I've implemented a model that can perform the forward pass of the tiny-yolov2 and I can say that in inputs you will have:
- inputs[:,:,:,:5] = t_x , t_y , t_w, t_h , t_objectness
- inputs[:,:,:,5:] = scores of the 20 classes
In your code you seem to take values from different positions, like the first value inputs[:,:,:,0] for 'iou' and also the others don't seem to match what the network is predicting. 

Can you please explain your choices? Thank you very much. By the way, have you been able to train the model?

**Comment by ss199302** [comment]
@TaihuLight  Hello, have you successfully implemented the test? Is the effect good?
